utils/python/CIME/test_scheduler.py

```diff
@@ -114,11 +114,8 @@ def __init__(self, test_names, test_data=None,
                        "Missing baseline comparison directory %s" % full_baseline_dir)
 
             # the following is to assure that the existing generate directory is not overwritten
-            if self._baseline_gen_name and not allow_baseline_overwrite:
+            if self._baseline_gen_name:
                 full_baseline_dir = os.path.join(self._baseline_root, self._baseline_gen_name)
-                expect(not os.path.isdir(full_baseline_dir),
```

Comment:
Because that is checking for the baseline root which can have a number of individual tests, we want to overwrite or not based on the existence of individual tests not baseline groups.
I do this in xml_phase